PWGJE/TableProducer/emcalClusterHadronicCorrectionTask.cxx

```diff
@@ -0,0 +1,309 @@
+        continue;
+      }
+
+      // Looping over all matched tracks for the cluster
+      // Total number of matched tracks = 20 (hard-coded)
+      for (const auto& match : tracksofcluster) {
+
+        double mom = abs(match.track_as<myTracks>().p());
+        registry.fill(HIST("h_matchedtracks"), 1);
+
+        // CASE 1: skip tracks with a very low pT
+        if (mom < 1e-6) {
+          continue;
+        } // end CASE 1
+
+        // CASE 2:
+        //  a) If one matched track -> 100% energy subtraction
+        //  b) If more than one matched track -> 100% energy subtraction
+        //  c) If you want to do systematic studies -> perform the above two checks a) and b), and then subtract 70% energy instead of 100%
+
+        // Perform dEta/dPhi matching
+        double dEta = match.track_as<myTracks>().eta() - cluster.eta();
+        double dPhi = TVector2::Phi_mpi_pi(match.track_as<myTracks>().phi() - cluster.phi());
+
+        // Apply the eta and phi matching thresholds
+        // dEta and dPhi cut : ensures that the matched track is within the desired eta/phi window
+
+        // Do pT-dependent track matching
+        if (doMomDepMatching) {
+          auto trackEtaMax = funcPtDepEta.Eval(mom);
+          auto trackPhiHigh = +funcPtDepPhi.Eval(mom);
+          auto trackPhiLow = -funcPtDepPhi.Eval(mom);
+
+          if ((dPhi < trackPhiHigh && dPhi > trackPhiLow) && fabs(dEta) < trackEtaMax) {
+            if (Nmatches == 0) {
+              closestTrkP = mom;
+            }
+            totalTrkP += mom;
+            Nmatches++;
+          }
+        } else {
+          // Do fixed dEta/dPhi matching (non-pT dependent)
+          if (fabs(dEta) >= minDEta || fabs(dPhi) >= minDPhi) {
+            continue; // Skip this track if outside the fixed cut region
+          }
+
+          // If track passes fixed dEta/dPhi cuts, process it
+          if (Nmatches == 0) {
+            closestTrkP = mom; // Closest track match
+          }
+          totalTrkP += mom; // Accumulate momentum
+          Nmatches++;       // Count this match
+        }
+
+      } // End of track loop
+      registry.fill(HIST("h2_ClsEvsNmatches"), cluster.energy(), Nmatches);
+
+      if (Nmatches >= 1) {
+        if (UseM02SubtractionScheme1) {
+          // Do M02-based correction if enabled
+          Ecluster1 = subtractM02ClusterEnergy(cluster.m02(), Ecluster1, Nmatches, closestTrkP, fHadCorr1, UseFraction1);
+          EclusterAll1 = subtractM02ClusterEnergy(cluster.m02(), EclusterAll1, Nmatches, totalTrkP, fHadCorralltrks1, UseFraction1);
+        } else {
+          // Default energy subtraction (100% and 70%)
+          Ecluster1 = subtractClusterEnergy(Ecluster1, closestTrkP, fHadCorr1, Nmatches, UseFraction1);
+          EclusterAll1 = subtractClusterEnergy(EclusterAll1, totalTrkP, fHadCorralltrks1, Nmatches, UseFraction1);
+        }
+
+        if (UseM02SubtractionScheme2) {
```

Comment:
what is the M02 subtraction scheme and what is the difference bewteen one and two?